src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -882,7 +882,9 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
             string[] dummyTargetsForEvaluationTime = _defaultTargetNames.Union(_targetNames).ToArray();
             foreach (string targetName in dummyTargetsForEvaluationTime)
             {
-                traversalProject.AddTarget(targetName);
+                ProjectTargetElement target = traversalProject.CreateTargetElement(targetName);
+                // Prepend so that any imported target overrides these default ones.
+                traversalProject.PrependChild(target);
```

Comment:
Unless it comes in via an sdk right? But that would mean an sdk in the before / after project files, which we could say it's not supported.

Reply:
What if you enhance ProjectTargetInstance to point to the previous target it overrides? Then you could massage the override chain via some internal APIs :)

Reply:
Or have an internal only method that hooks up a different AddTarget implementation in ProjectInstance which the solution parser hooks up to ignore the dummy targets https://github.com/microsoft/msbuild/blob/62fb89029dac66f73a5ca52363fd11b36f12190b/src/Build/Instance/ProjectInstance.cs#L1354